### PR TITLE
 Initial "infrastructure" for pack testing

### PR DESCRIPTION
**Pull request by Kami**
This pull request introduces some initial infrastructure for pack testing.

Most importantly, it adds a script for running all the tests for a specific pack. This script sets up the environment for a particular pack (installs all the dependencies, sets PYTHONPATH, etc.) and runs the tests.

Depends on https://github.com/StackStorm/st2/pull/2256

### CODE REVIEW

File: packs/docker/tests/test_sensor_docker_sensor.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        sensor.poll()
+        self.assertEqual(len(self.get_dispatched_triggers()), 1)
+        self.assertTriggerDispatched(trigger='docker.container_tracker.started',
+                                     payload={'container_info': MOCK_CONTAINER_DATA})
+
+        # One container stopped
+        sensor._get_active_containers.return_value = {}
+        sensor.poll()
+        self.assertEqual(len(self.get_dispatched_triggers()), 2)
+        self.assertTriggerDispatched(trigger='docker.container_tracker.stopped',
```
Comment (tonybaloney) [inline packs/docker/tests/test_sensor_docker_sensor.py]: :+1:

**Comment by dzimine** [comment]
Awesome.

**Comment by Kami** [comment]
Alright, this is ready for a review - feedback welcome.

File: Makefile [hunk 2]
```diff
@@ -42,13 +46,20 @@ metadata-check: requirements
 	@echo
 	. $(VIRTUALENV_DIR)/bin/activate; ${ROOT_DIR}/scripts/validate-pack-metadata-exists.sh
 
+.PHONY: .packs-tests
+.packs-tests:
+	@echo
+	@echo "==================== pylint ===================="
```
Comment (manasdk) [inline Makefile]: pack-tests :P

File: packs/csv/tests/test_action_parse.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+name1,surename1,1990
+""".strip()
+
+MOCK_DATA_2 = """
+first|last|year
+name1|surename1|1990
+""".strip()
+
+
+class ParseCSVActionTestCase(unittest2.TestCase):
```
Comment (manasdk) [inline packs/csv/tests/test_action_parse.py]: A `BaseActionTestCase` would be nice :)

Not sure if we have much value to add right now but I suspect we will in the future.

File: scripts/run-pack-tests.sh [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    echo "No tests found."
+    exit 0
+fi
+
+###################
+# Environment setup
+###################
+
+ST2_REPO_PATH=${ST2_REPO_PATH:-/tmp/st2}
+ST2_COMPONENTS=$(find ${ST2_REPO_PATH}/* -maxdepth 0 -name "st2*" -type d)
```
Comment (manasdk) [inline scripts/run-pack-tests.sh]: hmm, this is an interesting choice.

**Comment by tonybaloney** [comment]
@manasdk yes it would be nice, I've had to do some ugly things to simulate and mock out the action runner. https://github.com/tonybaloney/st2contrib/commit/33b20ed22d9c652716f4d7dfdfc94dbc1296c84f#diff-3b92aaffb116470df166266190848300R14 :+1:

File: packs/csv/tests/test_action_parse.py (hunk 3 above)
Comment (Kami) [inline packs/csv/tests/test_action_parse.py]: Yeah, I can add that and make it not do anything right now :)

**Comment by Kami** [comment]
@tonybaloney There is also this - https://github.com/StackStorm/st2/pull/2256/files#diff-95297328ad4514dde532bf959da7cb8fR31

**Pull request merged**